### PR TITLE
More multi-Z stuff (and removal of Crossed()/Uncrossed())

### LINKED ISSUE #1405: Revenant can't move z levels without piggybacking in a lift
What it says on the tin

### LINKED ISSUE #1811: Sentient Disease cannot change Z-levels for initial infection vector
## Round ID:
3793

In summary, for initial selection, you cannot select a target on a different Z-level, which largely limits your selection. This is particularly painful on the Aetherwhisp if you get the lower level, as you end up being stuck with a choice of medical or engineering, or on extremely rare occasions, security.

## Reproduction:
1. Become Sentient Disease
2. Attempt to switch Z-Levels to find a better target on initial selection
3. Be unable to click on ladders or have a button to switch Z-levels

### LINKED ISSUE #1893: Can't go down a Z-Level in zero gravity
Issue reported from  Round ID: 4140 (NSV 13 | Official | MRP | 98% Free Lag | Spaceship Combat Server)

Reporting client version: 514.1578

commits:

[#1797: 'More Sabre fixes' by covertcorvid at commit 7c9f2783b7](https://github.com/BeeStation/NSV13/pull/1797)
[#1801: 'overmap borders are more seamless for AI, and for shooting' by Ikalpo at commit afefd54de5](https://github.com/BeeStation/NSV13/pull/1801)
[#1846: 'Fixes some bad reference stuff with player ships' by Bokkiewokkie at commit 03a92238c2](https://github.com/BeeStation/NSV13/pull/1846)
[#1829: 'Self Surgery' by BetaCentari at commit 3bc3af01d1](https://github.com/BeeStation/NSV13/pull/1829)
[#1861: 'Snake Class medium gunboat' by Vasily2013 at commit a35f0a5846](https://github.com/BeeStation/NSV13/pull/1861)
[#1869: 'Knpc node next_id supports lists' by DeltaFire15 at commit 8b59d33d06](https://github.com/BeeStation/NSV13/pull/1869)
[#1840: 'Race Specific Emotes' by BetaCentari at commit 1d251a83e9](https://github.com/BeeStation/NSV13/pull/1840)
[#1830: 'Felinids don't like getting wet, and pulling them by the tail makes them upset' by red031000 at commit 4c1e4cfe6d](https://github.com/BeeStation/NSV13/pull/1830)
[#1878: 'WIP - Wire Guided Torpedoes' by Karmic-Skink at commit db836df7f9](https://github.com/BeeStation/NSV13/pull/1878)
[#1864: 'Removes NSV donator code' by covertcorvid at commit 4099bf77f1](https://github.com/BeeStation/NSV13/pull/1864)
[#1886: 'Implements spatial partitioning for physics collision detection' by TheFakeElon at commit 16734bd044](https://github.com/BeeStation/NSV13/pull/1886)
[#1876: 'Fixes Ammo Rack Control Console UI' by TheFakeElon at commit 71e7d50197](https://github.com/BeeStation/NSV13/pull/1876)
[#1848: 'Notoriety (aka NSV - Pulsar edition)' by DeltaFire15 at commit 7cb261214c](https://github.com/BeeStation/NSV13/pull/1848)
[#1890: 'Glock ammo crates give 5 mags instead of 1' by someone543 at commit 7155ba2592](https://github.com/BeeStation/NSV13/pull/1890)
[#1887: 'Ensmallens the starmap' by covertcorvid at commit f050e19658](https://github.com/BeeStation/NSV13/pull/1887)

Problem: Clicking a lower Z-Level in 0G appears to give you an option to go down, as with a more than 2 z level ladder, but instead plays a strange animation of your character slowly rising from the bottom z-level (I assume that is the intent) and it tells you that you're blocked by something.

Steps to reproduce:

1. Turn off grav gen.
2. Go over to an open square indicating a lower z-level such as the bottom of the stairs.
3. click the square.
4. A menu pops up, allowing you to close it or click a down arrow. this arrow is identical to multi-z-level ladders,
5. click the down arrow.
6. the game informs you that you are "floating up"
7. an animation of this plays.
8. At the end, it tells you that you are blocked by something,

### PR DESCRIPTION
**Pull request by covertcorvid**
## About The Pull Request
Several multi-Z fixes and improvements
Same thing as https://github.com/BeeStation/BeeStation-Hornet/pull/6724 and https://github.com/BeeStation/BeeStation-Hornet/pull/6858

Also ports https://github.com/BeeStation/BeeStation-Hornet/pull/5725

## Why It's Good For The Game
Fixes #1405 
Fixes #1811 
Fixes #1893 

## Changelog
:cl:
fix: fixed Bee's radial menu Z movement having some weird type requirements on the starting turf
tweak: made Move-Upwards and Move-Down use the animated travel_z proc for living things
fix: fixed sentient disease Z movement
fix: allowed most jaunt movement types to travel multi-Z regardless of gravity (includes revenants but not ninjas)
tweak: added a passthrough for attack_hand on stairs so the radial menu for multi-z travel would show up
/:cl:

### CODE REVIEW

**Comment by Pontenerd** [comment]
Question, why can't ninjas move between z levels?

**Comment by Karmic-Skink** [comment]
> Question, why can't ninjas move between z levels?

It is far too easy an evade, same reason ladders have a use timer.

**Comment by Pontenerd** [comment]
> > Question, why can't ninjas move between z levels?
> 
> Balance. They can still use stairs and ladders, but they are people and not ghosts

Yeah that seems fair enough

File: code/modules/mob/mob_movement.dm [hunk 1]
```diff
@@ -509,7 +515,11 @@
 /mob/proc/zMove(dir, feedback = FALSE)
 	if(dir != UP && dir != DOWN)
 		return FALSE
+	var/turf/source = get_turf(src)
```
Comment (TheFakeElon) [inline code/modules/mob/mob_movement.dm:518]: isn't loc being a turf a prerequisite for zMove being called

Comment (Bokkiewokkie) [reply code/modules/mob/mob_movement.dm:518]: Someone could be calling up() from inside a locker, bluespace body bag, or something else dumb no?

Comment (TheFakeElon) [reply code/modules/mob/mob_movement.dm:518]: I don't think they can

**Comment by covertcorvid** [comment]
DNM, pending mapping updates

**Review by Bokkiewokkie — APPROVED** [review]
I don't get why this whole connect_loc and connect_loc_behalf business needs to be so complicated and confusing, with a component, element and static lists of signals on top of using regular comsigs, but if it makes sense to you I trust it's fine.

**Comment by covertcorvid** [comment]
> I don't get why this whole connect_loc and connect_loc_behalf business needs to be so complicated and confusing, with a component, element and static lists of signals on top of using regular comsigs, but if it makes sense to you I trust it's fine.

It seems weirdly convoluted to me too but I'm sure someone had their reasons

**Pull request merged**